part.go

```diff
@@ -296,9 +329,11 @@ func parseParts(parent *Part, reader *bufio.Reader) error {
 		parent.AddChild(p)
 		if p.Boundary == "" {
 			// Content is text or data; build content reader pipeline.
-			if err := p.buildContentReaders(bbr); err != nil {
+			err := p.buildContentReaders(bbr)
+			if err != nil {
 				return err
 			}
+			p.Boundary = parent.Boundary
```

Comment:
I don't agree with this change; Boundary is a header field belonging to the parent object that it is defined on, it doesn't make sense to duplicate it on the children.  Especially not using the same Part field.

> if you want to retrieve the boundary of the html part inside the multipart/alternative you cant because the "parent" of this parts is its sibling txt.

As far as I know, your statement above is incorrect: siblings are not parents of each other. The Part.Parent should be pointing to the same parent (multipart/*) for all the siblings.   If that's not the case, then it sounds like a bug.

Let's remove this change and keep this PR scoped to the Clone and Test helper changes.  Then file a bug for the parent/boundary thing, possibly with some code or a unit test to illustrate the problem or use case.